src/places/PlaceManager.java

```diff
@@ -28,7 +28,7 @@
    * PlaceManager Attributes
    */
   private ArrayList<Place> places = new ArrayList<Place>();
-  private HashMap<String, Date> replicas = new HashMap<String, Date>();
+  private ConcurrentHashMap<String, Date> replicas = new ConcurrentHashMap<String, Date>();
```

Comment:
This is the core change. We will be using ConcurrentHashMap since it deals with concurrency at class level and not object or method level.
Basically, we could remove synchronized from all methods that access this object since the class will deal with it.